### PR TITLE
hdf5: Tests

### PR DESCRIPTION
**Pull request by jakirkham**
Appears we and the linter missed the tests in the PR. Would be nice to know why the linter isn't failing the build in that case. In any event, here are some tests to verify the libraries are all present.

### CODE REVIEW

File: recipes/hdf5/meta.yaml [hunk 1]
```diff
@@ -26,6 +26,22 @@ requirements:
   run:
     - zlib
 
+test:
```
Comment (pelson) [inline recipes/hdf5/meta.yaml]: If you packaged this into a single CLI, you'd make me very happy:

```
test:
   - assert_lib_exists hdf5 hd5_cpp hdf5_hl hd5_hl_cpp
```

**Comment by pelson** [comment]
```
Traceback (most recent call last):
  File "/Users/travis/miniconda/bin/conda-build-all", line 9, in <module>
    load_entry_point('conda-build-all==0.10.0', 'console_scripts', 'conda-build-all')()
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_build_all/cli.py", line 85, in main
    b.main()
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_build_all/builder.py", line 190, in main
    recipe_metas = self.fetch_all_metas()
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_build_all/builder.py", line 132, in fetch_all_metas
    recipe_metas = list_metas(conda_recipes_directory)
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_build_all/builder.py", line 75, in list_metas
    packages.append(MetaData(new_root))
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_build/metadata.py", line 340, in __init__
    self.parse_again(permit_undefined_jinja=True)
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_build/metadata.py", line 351, in parse_again
    self.meta = parse(self._get_contents(permit_undefined_jinja))
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_build/metadata.py", line 152, in parse
    (field, res[field].__class__.__name__))
RuntimeError: The test field should be a dict, not list
```

File: recipes/hdf5/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/hdf5/meta.yaml]: Sounds like a nice idea. Probably something nifty where we import this command from an external jinja template. At this point, just want to clear this out of `master` though.

**Comment by jakirkham** [comment]
Yep, missed `commands:`, which is now present.

File: recipes/hdf5/meta.yaml [hunk 2]
```diff
@@ -26,6 +26,43 @@ requirements:
...
+    - h5import --help
+    - h5jam --help
+    - h5ls --help
+    - h5mkgrp --help
+    - h5perf_serial --help
+    - h5redeploy --help
+    - h5repack --help
+    - h5repart --help
+    - h5stat --help
+    - h5unjam --help
```
Comment (jakirkham) [inline recipes/hdf5/meta.yaml]: I think these will play well on Windows, but I can't find docs for this ATM. We will see what AppVeyor says.

Comment (pelson) [inline recipes/hdf5/meta.yaml]: Looks like they are putting out exit codes (on circle)

Comment (jakirkham) [inline recipes/hdf5/meta.yaml]: Yeah, some of these don't have `--help` so we switched to using `command -v` on UNIX platforms and `where` on Windows.

File: recipes/hdf5/meta.yaml [hunk 3]
```diff
@@ -9,6 +9,14 @@ source:
...
   md5: b8ed9a36ae142317f88b0c7ef4b9c618
   patches:
+    # Patches the test suite to skip the `cache` test.
+    # This test has been found to rather resource intensive.
+    # In particular, it caused Travis CI's Mac builds to hang.
+    # Given that we simply skip the test on all platforms.
+    # For more details, see the PR where this was proposed.
+    #
+    # https://github.com/conda-forge/staged-recipes/pull/199
+    #
```
Comment (jakirkham) [inline recipes/hdf5/meta.yaml]: Added this explanation for our patch, as well.

File: recipes/hdf5/meta.yaml [hunk 4]
```diff
@@ -26,6 +34,49 @@ requirements:
...
+        "h5perf_serial",
+        "h5redeploy",
+        "h5repack",
+        "h5repart",
+        "h5stat",
+        "h5unjam"
+    ] %}
+    {% for each_hdf5_cmd in hdf5_cmds %}
+    - command -v {{ each_hdf5_cmd }}        # [unix]
+    - where {{ each_hdf5_cmd }}             # [win]
```
Comment (jakirkham) [inline recipes/hdf5/meta.yaml]: Now tests with `command -v` and `where` as some commands don't have `--help`.

**Comment by jakirkham** [comment]
Circle CI is now happy. The linter says it is [happy]( https://travis-ci.org/conda-forge/staged-recipes/jobs/118417863#L256 ). Hoping Travis CI and AppVeyor will like this.

Going to call it for a bit.

**Comment by pelson** [comment]
I'm just going to merge and iterate. Thanks @jakirkham

**Comment by jakirkham** [comment]
Thanks @pelson. Hopefully, it doesn't give more grief. Will check back in awhile to see if it went ok.

**Comment by jakirkham** [comment]
FWIW, I did a local build on my Mac and it worked just fine. So, AppVeyor/Windows is the only outstanding question in my mind at least.

File: recipes/hdf5/meta.yaml (hunk 3 above)
Comment (ocefpaf) [inline recipes/hdf5/meta.yaml]: Nice. But we can skip it on `# [os x ]` only, no?

**Pull request merged**